assemble/Advection_Diffusion_DG.F90

```diff
@@ -819,18 +838,114 @@ subroutine construct_advection_diffusion_dg(big_m, rhs, field_name,&
     call allocate(U_nl, U_nl_backup%dim, U_nl_backup%mesh, "LocalNonlinearVelocity")
     call set(U_nl, U_nl_backup)
 
-
-    Diffusivity=extract_tensor_field(state, trim(field_name)//"Diffusivity"&
+    Diffusivity_ptr => extract_tensor_field(state, trim(field_name)//"Diffusivity"&
          &, stat=stat)
     if (stat/=0) then
-       call allocate(Diffusivity, T%mesh, trim(field_name)//"Diffusivity",&
+       call allocate(Diffusivity, T%mesh, "Diffusivity",&
             FIELD_TYPE_CONSTANT)
        call zero(Diffusivity)
        include_diffusion=.false.
     else
-       ! Grab an extra reference to cause the deallocate below to be safe.
-       call incref(Diffusivity)
+       call allocate(Diffusivity, T%mesh, "Diffusivity")
+       call set(Diffusivity, Diffusivity_ptr)
```

Comment:
Do we really need to copy the entire diffusivity tensor even if we don't have LES?